action.yaml

```diff
@@ -449,6 +453,38 @@ runs:
         echo ::endgroup::
       shell: bash
       id: deploy-options
+    - name: Determine if Development Mode is enabled
+      if: ${{ inputs.wake-on-deploy == 'true' && inputs.action != 'delete-deployment-preview' }}
+      run: |
+        echo ::group::Determine if Development Mode is enabled
+        DEPLOYMENT_TYPE=$(astro deployment inspect ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --clean-output --key configuration.deployment_type)
+        # only inspect development mode if deployment is not hybrid
+        if [[ $DEPLOYMENT_TYPE != "HYBRID" ]]; then
+          DEV_MODE=$(astro deployment inspect ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --clean-output --key configuration.is_development_mode)
+          echo "Deployment development mode: $DEV_MODE"
+          if [[ $DEV_MODE == "" ]]; then
+            echo "DEVELOPMENT_MODE=false" >> $GITHUB_OUTPUT
+          else
+            echo "DEVELOPMENT_MODE=$DEV_MODE" >> $GITHUB_OUTPUT
+          fi
+        else
+          echo "DEVELOPMENT_MODE=false" >> $GITHUB_OUTPUT
+        fi
+        echo ::endgroup::
+      shell: bash
+      id: development-mode
+    - name: Override to wake up the Deployment
+      if: ${{ inputs.wake-on-deploy == 'true' && inputs.action != 'delete-deployment-preview' && steps.development-mode.outputs.DEVELOPMENT_MODE == 'true' }}
+      run: |
+        echo ::group::Override to wake up the Deployment
+        astro deployment wake-up ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --force
```

Comment:
not related to deploy-action, but just to be double sure, let's check that the wake-up command correctly returns exit code 1 on failure, otherwise we might end up in an endless loop 😅